raiden_libs/test/mocks/client.py

```diff
@@ -57,16 +59,28 @@ def __init__(
         self,
         privkey: str,
         token_network_contract: Contract,
-        token_contract: Contract
+        token_contract: Contract,
     ) -> None:
         self.privkey = privkey
         self.address = private_key_to_address(privkey)
         self.contract = token_network_contract
         self.token_contract = token_contract
         self.partner_to_channel_id: Dict[Address, ChannelIdentifier] = dict()
         self.token_network_abi = None
+        self.transport: Transport = None
         self.client_registry: Dict[Address, 'MockRaidenNode'] = dict()
         self.web3 = self.contract.web3
+        if self.transport is not None:
+            self.transport.start()  # type: ignore
+            self.transport.add_message_callback(self.on_message_event)  # type: ignore
+
+    def on_message_event(self, message: Message):
```

Comment:
What is this for? Can we get around adding a greenlet here?

Reply:
Client has to receive and process the messages from PFS somehow. Any suggestions?

Reply:
What do you want to do with this message?

If you want to test that the PFS sends the right answer, then mocking the transport on PFS side  should be enough.

Reply:
I think at some point the client would like to process this message, as it requested it in the first place. So I think it makes sense to add methodology here, also for future work. Mocking it on PFS side isn't a lot less code, but makes no sense w.r.t. future functionality.

Reply:
The whole point of the MockClient is to create fake message for testing, and these messages are not send over matrix itself but over a DummyTransport at the PFS side. So I don't see the point in changing this here.

Reply:
We have this argument because you mix two different tests. One is testing that the reply from the PFS is correct, and the other is testing that the transport works.

Testing the transport will happen later, so for now we just want to check for a proper response. And for this we don't need this stuff.

Reply:
Well you didn't answer the question. This is not about two different tests, but about writing code that will be disposed soon after: What you're suggesting is doing something like 
`pathfinding_service.transport.transmit(request_paths)`. This my be syntactically correct, but doesn't make any sense semantically; the PFS will never send `request_paths` messages to it self, the client will send them. For me writing tests is not about getting some green lights as fast as possible, but instead about writing and testing functionality that will later be used. In our case this are the client and pfs methods to request, send, process and reply paths respectively. 
However, I'll do a separate PR for client transport to finally get this one merged.